corev_apu/tb/ariane_tb.cpp

```diff
@@ -335,15 +337,23 @@ int main(int argc, char **argv) {
   top->rst_ni = 1;
 
   // Preload memory.
-  size_t mem_size = 0xFFFFFF;
 #if (VERILATOR_VERSION_INTEGER >= 5000000)
   // Verilator v5: Use rootp pointer and .data() accessor.
-  memif.read(0x80000000, mem_size, (void *)top->rootp->ariane_testharness__DOT__i_sram__DOT__gen_cut__BRA__0__KET____DOT__i_tc_sram_wrapper__DOT__i_tc_sram__DOT__sram.m_storage);
+#define MEM top->rootp->ariane_testharness__DOT__i_sram__DOT__gen_cut__BRA__0__KET____DOT__i_tc_sram_wrapper__DOT__i_tc_sram__DOT__sram.m_storage
 #else
   // Verilator v4
-  memif.read(0x80000000, mem_size, (void *)top->ariane_testharness__DOT__i_sram__DOT__gen_cut__BRA__0__KET____DOT__i_tc_sram_wrapper__DOT__i_tc_sram__DOT__sram);
+#define MEM top->ariane_testharness__DOT__i_sram__DOT__gen_cut__BRA__0__KET____DOT__i_tc_sram_wrapper__DOT__i_tc_sram__DOT__sram
 #endif
-  // memif.read(0x84000000, mem_size, (void *)top->ariane_testharness__DOT__i_sram__DOT__gen_cut__BRA__0__KET____DOT__gen_mem__DOT__gen_mem_user__DOT__i_tc_sram_wrapper_user__DOT__i_tc_sram__DOT__sram);
+  long long addr;
+  long long len;
+
+  size_t mem_size = 0xFFFFFF;
+  while(get_section(&addr, &len))
+  {
+    read_section_void(addr, (void *) MEM , mem_size);
+    if (addr == 0x80000000)
```

Comment:
I have done this code because previously it was only loading the memory this particular memory address. If an iteration is done to the sections it would cause a non-ending execution. as it will override always from this particular address. I will try to make it prettier and if it does not convince you I will revert the change